lib/Doctrine/ORM/Mapping/Driver/AnnotationDriver.php

```diff
@@ -340,6 +340,7 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
                 $mapping['cascade'] = $oneToOneAnnot->cascade;
                 $mapping['orphanRemoval'] = $oneToOneAnnot->orphanRemoval;
                 $mapping['fetch'] = $this->getFetchMode($className, $oneToOneAnnot->fetch);
+                $mapping['indexName'] = $oneToOneAnnot->indexName;
```

Comment:
@zeroedin-bill I disagree here. It's a lot easier and less error prone to have an attribute on the association simply for the index name. Also it relates a lot better to have it at this place rather than loosely in the table section. You can still do your own `@Index` definitions to overwrite implicit indexes if you want to. But simply being able to change the index name without having to clutter the definition with another `@Index` makes totally sense IMO.